internal/indexcoord/index_coord.go

```diff
@@ -854,6 +863,46 @@ func (i *IndexCoord) watchMetaLoop() {
 	}
 }
 
+func (i *IndexCoord) tryAcquireSegmentReferLock(ctx context.Context, segIDs []UniqueID) error {
+	status, err := i.dataCoordClient.AcquireSegmentLock(ctx, &datapb.AcquireSegmentLockRequest{
+		NodeID:     i.session.ServerID,
+		SegmentIDs: segIDs,
+	})
+	if err != nil {
+		log.Error("IndexCoord try to acquire segment reference lock failed", zap.Int64s("segIDs", segIDs),
+			zap.Error(err))
+		return err
+	}
+	if status.ErrorCode != commonpb.ErrorCode_Success {
+		log.Error("IndexCoord try to acquire segment reference lock failed", zap.Int64s("segIDs", segIDs),
+			zap.Error(errors.New(status.Reason)))
+		return errors.New(status.Reason)
+	}
+	return nil
+}
+
+func (i *IndexCoord) tryReleaseSegmentReferLock(ctx context.Context, segIDs []UniqueID) {
+	releaseLock := func() error {
+		status, err := i.dataCoordClient.ReleaseSegmentLock(ctx, &datapb.ReleaseSegmentLockRequest{
+			NodeID:     i.session.ServerID,
+			SegmentIDs: segIDs,
+		})
+		if err != nil {
+			return err
+		}
+		if status.ErrorCode != commonpb.ErrorCode_Success {
+			return errors.New(status.Reason)
+		}
+		return nil
+	}
+	err := retry.Do(ctx, releaseLock, retry.Attempts(100))
+	if err != nil {
+		log.Error("IndexCoord try to release segment reference lock failed", zap.Int64s("segIDs", segIDs),
+			zap.Error(err))
+		panic(err)
```

Comment:
no need to panic but return error~

Reply:
post execute should return handle it. 
So far we post execute could not handle error, so panic if retry failed

	defer func() {
		//task postExecute
		span.LogFields(oplog.Int64("processTask: scheduler process PostExecute", t.getTaskID()))
		t.postExecute(ctx)
	}()

Reply:
the post execute definition should remove err because it can not be handled

	postExecute(ctx context.Context) error